mobile/examples/phi-3/android/app/src/main/cpp/native-lib.cpp

```diff
@@ -0,0 +1,187 @@
+#include <jni.h>
+#include <string>
+#include <vector>
+
+#include <android/log.h>
+
+#include "ort_genai_c.h"
+
+namespace {
+    void ThrowException(JNIEnv *env, OgaResult *result) {
+        __android_log_write(ANDROID_LOG_DEBUG, "native", "ThrowException");
+        // copy error so we can release the OgaResult
+        jstring jerr_msg = env->NewStringUTF(OgaResultGetError(result));
+        OgaDestroyResult(result);
+
+        static const char *className = "ai/onnxruntime/genai/demo/GenAIException";
+        jclass exClazz = env->FindClass(className);
+        jmethodID exConstructor = env->GetMethodID(exClazz, "<init>", "(Ljava/lang/String;)V");
+        jobject javaException = env->NewObject(exClazz, exConstructor, jerr_msg);
+        env->Throw(static_cast<jthrowable>(javaException));
+    }
+
+    void ThrowIfError(JNIEnv *env, OgaResult *result) {
+        if (result != nullptr) {
+            ThrowException(env, result);
+        }
+    }
+
+    // handle conversion/release of jstring to const char*
+    struct CString {
+        CString(JNIEnv *env, jstring str)
+                : env_{env}, str_{str}, cstr{env->GetStringUTFChars(str, /* isCopy */ nullptr)} {
+        }
+
+        const char *cstr;
+
+        operator const char *() const { return cstr; }
+
+        ~CString() {
+            env_->ReleaseStringUTFChars(str_, cstr);
+        }
+
+    private:
+        JNIEnv *env_;
+        jstring str_;
+    };
+}
+
+extern "C" JNIEXPORT jlong JNICALL
+Java_ai_onnxruntime_genai_demo_GenAIWrapper_loadModel(JNIEnv *env, jobject thiz, jstring model_path) {
+    CString path{env, model_path};
+    __android_log_print(ANDROID_LOG_DEBUG, "native", "loadModel %s", path.cstr);
```

Comment:
We could probably remove a lot of this low value output. It was intended to debug the initial code but now that everything is working is most likely not needed.

I'd suggest leaving a few print statements around the token generation as that can be useful and would provide an example of adding extra debug output if needed.